lib/services/android-project-service.ts

```diff
@@ -118,19 +121,134 @@ class AndroidProjectService implements IPlatformProjectService {
 			return assetsDirectory;
 
 		}).future<string>()();
-	}
+    }
+
+    private updateMetadata(projectRoot: string): void {
+        var projMetadataDir = path.join(projectRoot, "assets", "metadata");
+        var libsmetadataDir = path.join(projectRoot, "../../lib", this.platformData.normalizedPlatformName, AndroidProjectService.METADATA_DIRNAME);
+        shell.cp("-f", path.join(libsmetadataDir, "*.dat"), projMetadataDir);
+    }
+
+    private generateMetadata(projectRoot: string): void {
+        var metadataGeneratorPath = path.join(__dirname, "../../resources/tools/metadata-generator.jar");
+        var libsFolder = path.join(projectRoot, "../../lib", this.platformData.normalizedPlatformName);
+		var metadataDirName = AndroidProjectService.METADATA_DIRNAME;
+        var inputDir = libsFolder;
```

Comment:
Do you write program for the compiler/computer or for the humans? As suggested https://github.com/fealebenpae/nativescript-cli/commit/e8aac716ef5a8e48a0b69a6cd18235113b4967b5